### PR TITLE
feat: Add {first name} variable to Organizer & Scheduler custom event…

### LINKED ISSUE #13974: [CAL-3302] [CAL-3258] Add {first name} variable to Organizer & Scheduler custom event name
Currnetly we can only do full name. Would be nice to do first name if you have split the name collection in the booking questions.

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3258](https://linear.app/calcom/issue/CAL-3258/add-first-name-variable-to-organizer-and-scheduler-custom-event-name)</sub>

<sub>[CAL-3302](https://linear.app/calcom/issue/CAL-3302/[cal-3258]-add-first-name-variable-to-organizer-and-scheduler-custom)</sub>

### PR DESCRIPTION
**Pull request by Amit91848**
## What does this PR do?

Fixes #13974

https://github.com/calcom/cal.com/assets/74371312/b039f9d2-e521-458c-acf6-b638ed7464ad

## Type of change

- New feature (non-breaking change which adds functionality)

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

**Review by CarinaWolli — DISMISSED** [review]
Like mentioned in the issue, this should use the first name when split name is enabled. 

Split name can be enabled in, Booking questions -> Advanced -> Booking question -> Edit name: 
Screenshot 2024-04-19 at 5 21 17 PM

We should add {Scheduler first name} and {Scheduler last name} and use the values from the booking question

**Comment by Amit91848** [comment]
Should we only show those variables when split full name is checked? Or should we add some helper text that it will only work when you have set it to true?

**Comment by CarinaWolli** [comment]
> Should we only show those variables when split full name is checked? Or should we add some helper text that it will only work when you have set it to true?

It would be best to only show them if split name is enabled

**Comment by Amit91848** [comment]
Made changes @CarinaWolli. Also the issue was closed, is this still relevant?

**Comment by CarinaWolli** [comment]
I reopened the issue, I think it's still nice to have

File: packages/core/event.ts [hunk 1]
```diff
@@ -57,11 +57,25 @@ export function getEventName(eventNameObj: EventNameObjectType, forAttendeeView
...
+    typeof eventNameObj.bookingFields.name === "object" &&
+    !Array.isArray(eventNameObj.bookingFields.name) &&
+    eventNameObj.bookingFields.name.firstName
+  ) {
+    dynamicEventName = dynamicEventName.replaceAll(
+      "{Scheduler first name}",
+      eventNameObj.bookingFields.name.firstName.toString()
+    );
+  }
+
```
Comment (Udit-takkar) [inline packages/core/event.ts:78]: NIT
```suggestion
  if (
    eventNameObj.bookingFields?.name &&
    typeof eventNameObj.bookingFields.name === "object" &&
    !Array.isArray(eventNameObj.bookingFields.name) &&
    eventNameObj.bookingFields.name.firstName
  ) {
    dynamicEventName = dynamicEventName.replaceAll(
      "{Scheduler first name}",
      eventNameObj.bookingFields.name.firstName.toString()
    );
  }

```

and creating a func isObj to check typeof eventNameObj.bookingFields.name === "object" &&
    !Array.isArray(eventNameObj.bookingFields.name) &&

Comment (Amit91848) [reply packages/core/event.ts]: Screenshot 2024-05-13 at 2 19 10 PM

Cannot shake this error

**Review by Udit-takkar — DISMISSED** [review]
LGTM. pushed some changes.

File: packages/core/event.ts [hunk 2]
```diff
@@ -105,6 +113,8 @@ export const validateCustomEventName = (
     "{Organiser}",
     "{Scheduler}",
     "{Location}",
+    "{Organiser first name}",
```
Comment (CarinaWolli) [inline packages/core/event.ts:116]: as mentioned in https://github.com/calcom/cal.com/pull/14156#pullrequestreview-2012481181, we wanted `{Scheduler first name}` and `{Scheduler last name}`

But you can leave `{Organiser first name}`, that's fine

**Review by CarinaWolli — CHANGES_REQUESTED** [review]
See comment https://github.com/calcom/cal.com/pull/14156#discussion_r1613634339

File: packages/core/event.ts (hunk 2 above)
Comment (Amit91848) [reply packages/core/event.ts:116]: The issue stated first name, which is why I did not include last name. I will make changes to it.

**Review by CarinaWolli — APPROVED** [review]
Nice work, looks good 👏🏻

**Pull request merged**